src/evaluate.jl

```diff
@@ -88,8 +97,25 @@ function evaluate{T<:Number}(a::Taylor1{T}, x::Taylor1{T})
     end
     suma
 end
+function evaluate{T<:NumberNotSeries}(a::Taylor1{Taylor1{T}}, x::Taylor1{T})
```

Comment:
`NumberNotSeries` here may be too restrictive; I'm thinking in cases such as `Taylor1{Taylor1{...{T}...}`. Would everything work if `T<:Number`?

Reply:
Sure, it should work! Thanks!